handlers/meshery_pattern_handler.go

```diff
@@ -148,14 +64,115 @@ func (h *Handler) SavePatternFile(
 		return
 	}
 
-	resp, err := provider.SaveMesheryPattern(token, parsedBody)
-	if err != nil {
-		http.Error(rw, fmt.Sprintf("failed to save the pattern: %s", err), http.StatusInternalServerError)
+	format := r.URL.Query().Get("output")
+
+	// If Content is not empty then assume it's a local upload
+	if parsedBody.PatternData != nil {
+		patternName, err := models.GetPatternName(parsedBody.PatternData.PatternFile)
```

Comment:
@utkarsh-pro CMIIW but I suppose this line should be 
`patternName, err := models.GetPatternName(parsedBody.PatternData.Name)` as we are trying to get the name of the file from model.MesheryPattern.

Reply:
The code is correct. GetPatternName will read the YAML to extract the name.